src/AutoRest.CSharp/Common/Output/Models/Requests/RestClientMethod.cs

```diff
@@ -73,6 +73,14 @@ accessibility switch
 
         public ResponseClassifierType ResponseClassifierType { get; }
 
-        public PropertyBag? PropertyBag { get;  }
+        public PropertyBag? PropertyBag { get; }
+
+        public bool ShouldEnableRedirect => Responses.Any(r => r.StatusCodes.Any(r => IsRedirectResponseCode(r.Code)));
+
+        private bool IsRedirectResponseCode(int? code) => code switch
+        {
+            300 or 301 or 302 or 303 or 307 or 308 => true,
```

Comment:
Do we have any guideline that we only consider those codes as redirection?

Reply:
- I checked the semantic meaning of each 30X response.
- Also refer to `autorest.testserver`: https://github.com/Azure/autorest.testserver/blob/main/swagger/httpInfrastructure.json